### DISCUSSION TITLE
Radar target detected out of fov

### DISCUSSION THREAD

**Discussion by ShuffleWire** [cyberbotics/webots#4625] [Ideas]
**Describe the Bug**
Radar object could be detected even if they are out of the radar fov. Additionally the heading of the (ghost) target is clamped between min and max angles value of radar fov, which make it even more wrong.
It's caused by the absence of boundingObject, in particular when I use only one Solid, not inside a robot (contrary to the "radar" sample device, which define radar target inside Mybot2/3/4, and the boundingBox of these seems to make the thing work somehow)

**Steps to Reproduce**
1. Open and run the following scene [radar.zip](https://github.com/cyberbotics/webots/files/8078832/radar.zip)

2. MyBot run, initially see the target, it's fine
3. Then the target go out of the fov, and the azimut stuck to 0.39, and the target is still "detected"

**Expected behavior**
The target should not be seen, and wb_radar_get_number_of_targets() should return 0

**System**
 - Operating System: Debian 11
 - Webots: 2022a and develop b5e7ec85c1e63a0a07febe87af8b7e3ce0725b2c

**Additional context**
Afaik, there should be something with the recursivelyComputeBounds which work when the target is embedded in another Solid.
But in my case, in computeBounds, boundingObject == NULL, hence nodeType stay at WB_NODE_NO_NODE, hence messing up the detection.

Note that adding an bounding box to the "Cylinder" in my example solve the issue, but I don't feel right to make that, and it would interfere with the rest of the scene during collision handling, in more complex scene (and add useless performance cost)

**Comment by ad-daniel** [comment]
Indeed it appears to be a bug, as you say defining a `boundingObject` works as a temporary solution, but it shouldn't be required to. Thank you for reporting it.

**Comment by ad-daniel** [comment]
Actually the cause is that since no bounding object has been defined, it uses the bounding sphere as bounds for the detection and since this object is a very tall cylinder, the bounding sphere is a sphere of diameter equal to the height of the cylinder, which will of course be visible by the radar as it's huge. Changing the height to 0.1 for instance, the detection stops when leaving the FOV.
So I'm not sure what is the best way of solving this, it uses the bounding sphere precisely because in absence of the bounding object this is the safest option, but perhaps the behavior should be different in case of primitive shapes such as this.
